[BUG]KubeBlocks Controller Crashes Due to Zero-Prefixed Version Numbers

## Issue Description

The KubeBlocks controller pod is continuously crashing and restarting, preventing normal operation. Log analysis reveals that the controller panics when processing version numbers with leading zeros.

## Steps to Reproduce

1. Create a ComponentVersion resource containing a version number with leading zeros, such as "2025.03.0"
2. Observe the KubeBlocks controller pod repeatedly restarting
3. Check the pod logs to see the following panic:

```
panic: runtime error - invalid service version in comparator: illegal zero-prefixed version component "03" in "2025.03.0"
```

## Detailed Error Log

```
2025-05-16T08:20:53.005Z	INFO	Observed a panic in reconciler: runtime error - invalid service version in comparator: illegal zero-prefixed version component "03" in "2025.03.0"	{"controller": "componentversion", "controllerGroup": "apps.kubeblocks.io", "controllerKind": "ComponentVersion", "ComponentVersion": {"name":"neo4j"}, "namespace": "", "name": "neo4j", "reconcileID": "9bb70817-9987-428c-8dd6-2a9b8a8807a6"}
panic: runtime error - invalid service version in comparator: illegal zero-prefixed version component "03" in "2025.03.0" [recovered]
	panic: runtime error - invalid service version in comparator: illegal zero-prefixed version component "03" in "2025.03.0"
```

The error occurs in the `serviceVersionComparator` function at `controllers/apps/componentversion_controller.go:406`.

## Environment Information

```
kbcli version
Kubernetes: v1.28.3
KubeBlocks: 1.0.0-beta.47
kbcli: 1.0.0-beta.21
```

## Expected Behavior

The version comparator should properly handle version numbers with leading zeros, such as "2025.03.0".

## Temporary Workaround

Modify the ComponentVersion resource to change the version number from "2025.03.0" to "2025.3.0" (removing the leading zero).

## Suggested Fix

Update the version comparison logic to properly handle version number components with leading zeros, or clearly reject versions with leading zeros during validation with a clear error message, rather than causing the controller to crash.

--- a/controllers/apps/componentversion_controller.go
+++ b/controllers/apps/componentversion_controller.go
@@ -254,7 +254,7 @@ func (r *ComponentVersionReconciler) status(cli client.Client, rctx intctrlutil.
 func (r *ComponentVersionReconciler) supportedServiceVersions(compVersion *appsv1.ComponentVersion) string {
 	versionSet := sets.New[string]()
 	for _, release := range compVersion.Spec.Releases {
-		if len(release.ServiceVersion) > 0 {
+		if err := validateServiceVersion(release.ServiceVersion); err == nil {
 			versionSet.Insert(release.ServiceVersion)
 		}
 	}
@@ -401,14 +401,8 @@ func serviceVersionComparator(a, b string) int {
 	if len(b) == 0 {
 		return 1
 	}
-	v, err1 := version.ParseSemantic(a)
-	if err1 != nil {
-		panic(fmt.Sprintf("runtime error - invalid service version in comparator: %s", err1.Error()))
-	}
-	ret, err2 := v.Compare(b)
-	if err2 != nil {
-		panic(fmt.Sprintf("runtime error - invalid service version in comparator: %s", err2.Error()))
-	}
+	v, _ := version.ParseSemantic(a)
+	ret, _ := v.Compare(b)
 	return ret
 }
 

